### ISSUE TITLE
keybindings not work with vi mode in fish

### ISSUE THREAD

**Issue by explosic4** [junegunn/fzf#468]
Keybindings work with the default mode in fish, but not with the vi mode. 
However I can use keybindings with vi mode after I executed function `fzf_key_bindings` in fish shell. If I open another shell, I need to do it again. 
I tried adding `fzf_key_bindings` in the `config.fish`, so this function can execute automatically when I startup fish, but the keybindings still does not work.

**Comment by junegunn** [comment]
Hi, vi mode was added to fish pretty recently, right? Unfortunately I personally don't use fish and don't have enough experience or knowledge on fish. Looking for contribution.

**Comment by explosic4** [comment]
It looks like that in vi mode the keybindings  in `fzf_key_bindings.fish` will be overwritten by vi mode keybindings or just make no effect, I am not sure. Anyway I can use keybindings by executing `fzf_key_bindings' in fish shell after launching it.

**Comment by junegunn** [comment]
Can you share your configuration file?

**Comment by explosic4** [comment]
```
# config.fish

set fisher_home ~/SourcePackages/fisherman
set fisher_config ~/.config/fisherman
source $fisher_home/config.fish

fish_vi_mode
```
I use [fisherman](https://github.com/fisherman/fisherman) to manage the plugins for fish, i only installed [theme-agnoster](https://github.com/oh-my-fish/theme-agnoster), [fasd](https://github.com/oh-my-fish/plugin-fasd) and [pkg-fasd](https://github.com/gretel/pkg-fzf)

**Comment by junegunn** [comment]
Related: https://github.com/fish-shell/fish-shell/issues/2254

So probably this will be fixed in the next release of fish. The comments in the issue imply that the user key bindings should be written with both emacs and vi mode in mind. But fortunately fzf key bindings already work just as expected in both modes.

**Comment by dpc** [comment]
From my quick investigation it seems that (`.config/fish/config.fish`):

```
set fish_key_bindings fish_vi_key_bindings
```

Is changing the function that is going to be called to set keybindings. It was previously `fish_user_key_bindings` and FZF installed it's bind hook in `.config/fish/functions/fish_user_key_bindings.fish`, but it's not going to be ignored.

As a workaround I've moved the bindings into `.config/fish/functions/fish_prompt.fish`. It can be outside of the `function fish_prompt` and still works.

I'm doing some personal trickery (haven't used fzf install script), so just for reference I have appened:

```
source .fzf/shell/key-bindings.fish
fzf_key_bindings
```

**Comment by riclima** [comment]
I found a simple solution, which seems to me like the proper way to configure using both the fzf keybindings and fish vi keybindings. I added `fish_vi_key_bindings` to the file `~/.config/fish/functions/fish_user_key_bindings.fish`, before the `fzf_key_bindings` line. My file looks like this:

    fish_vi_key_bindings
    fzf_key_bindings

Note that this is on fish 2.2.0.

I think this is not an fzf issue—just a quirk of how configuration works in fish. I suggest adding a note  in the `README` file in the Tips section mentioning this for people who want to use the vi key bindings and close the issue.
